db/sql/00_msar.sql

```diff
@@ -948,6 +948,54 @@ FROM (
 $$ LANGUAGE SQL STABLE;
 
 
+CREATE OR REPLACE FUNCTION msar.get_role(rolename text) RETURNS jsonb AS $$/*
+Given a rolename, return a JSON object describing the role in a database server.
+
+The returned JSON object has the form:
+  {
+    "oid": <int>
+    "name": <str>
+    "super": <bool>
+    "inherits": <bool>
+    "create_role": <bool>
+    "create_db": <bool>
+    "login": <bool>
+    "description": <str|null>
+    "members": <[
+        { "oid": <int>, "admin": <bool> }
+      ]|null>
+  }
+*/
+WITH rolemembers as (
+  SELECT
+    pgr.oid AS oid,
+    jsonb_agg(
+      jsonb_build_object(
+        'oid', pgm.member::bigint,
+        'admin', pgm.admin_option
+      )
+    ) AS members
+  FROM pg_catalog.pg_roles pgr
+    INNER JOIN pg_catalog.pg_auth_members pgm ON pgr.oid=pgm.roleid
+  GROUP BY pgr.oid
+)
+SELECT jsonb_build_object(
+  'oid', r.oid::bigint,
+  'name', r.rolname,
+  'super', r.rolsuper,
+  'inherits', r.rolinherit,
+  'create_role', r.rolcreaterole,
+  'create_db', r.rolcreatedb,
+  'login', r.rolcanlogin,
+  'description', pg_catalog.shobj_description(r.oid, 'pg_authid'),
+  'members', rolemembers.members
+)
+FROM pg_catalog.pg_roles r
+  LEFT OUTER JOIN rolemembers ON r.oid = rolemembers.oid
+WHERE r.rolname = rolename;
+$$ LANGUAGE SQL STABLE;
```

Comment:
We already have a `msar.get_roles` function which has the same query & json response. Can we modify the function arguments to accept an optional filter to add to the where clause and re-use the same instead of duplicating it?

@mathemancer What are your thoughts on the approach?